src/Pages/EmailList/EmailTemplate.js

```diff
@@ -0,0 +1,294 @@
+    super(props);
+    this.state = {
+      events: [],
+      recipients: '',
+      subject: '',
+      loadedContent: '',
+      content: '',
+      users: '',
+      memberEmail: [],
+      officerEmail: [],
+      allEmail: [],
+    };
+  }
+
+  // When page loads, perform getUsers() for email lists && getEvents() for
+  // events in template
+  componentDidMount() {
+    if (this.props.user) {
+      this.getUsers();
+    }
+    this.getEvents();
+  }
+
+  getUsers = async () => {
+    let apiResponse = await getAllUsers(this.props.user.token);
+    if (!apiResponse.error) {
+      this.setState({ users: apiResponse.responseData });
+    }
+    // after getting api response, store them in state vars with function
+    this.getFilteredUsers();
+  };
+
+  getFilteredUsers = () => {
+    let allUsers = filterUsers(this.state.users, 0);
+    let officerEmail = [],
+      memberEmail = [],
+      allEmail = [];
+    // forEach loop to push emails to their respective arrays
+    allUsers.forEach(function(item) {
+      // Access level >=2 is officers+, Access level >=0 but <2 are members
+      if (item.accessLevel >= 2) {
+        allEmail.push(item.email);
+        officerEmail.push(item.email);
+      } else if (item.accessLevel >= 0) {
+        allEmail.push(item.email);
+        memberEmail.push(item.email);
+      }
+    }, this.setState({ officerEmail, memberEmail, allEmail }));
+  };
+
+  getEvents = async () => {
+    const apiResponse = await getAllEvents();
+    if (!apiResponse.error) {
+      this.storeContent(apiResponse);
+    } else {
+      let errorMessage = 'Unable to retrieve events.';
+      this.setState({ events: [errorMessage] });
+    }
+  };
+
+  // customize data with html, append header/footer, and store that to 'content'
+  storeContent = async (events) => {
+    const mailHeader =
+      '<p><span style="font-size: 10pt; font-family: Arial;">' +
+      '<strong>Hello SCE G4NG,</strong></span></p>' +
+      '\n<p><span style="font-size: 10pt; font-family: Arial;">' +
+      '<strong>This is our newsletter, please read ty.</strong></span></p>\n';
+
+    const mailFooter =
+      '<p><strong><span style="color: #236fa1; font-size: 10pt;">Software ' +
+      'and Computer Engineering</span><span style="color: #236fa1; ' +
+      'font-size: 10pt;"> Society<br /></span></strong>' +
+      '<span style="font-size: 10pt;">Website: ' +
+      '<a href="http://sce.engr.sjsu.edu">http://sce.engr.sjsu.edu</a><br />' +
+      'Contact: <a href="mailto:sce.sjsu@gmail.com">sce.sjsu@gmail.com</a>' +
+      '<br />Facebook: <a href="https://www.facebook.com/sjsusce/">@scesjsu' +
+      '</a><br />Instagram: <a href="https://www.instagram.com/sjsusce/">' +
+      '@scesjsu</a><br />CmpE Slack: <a href="https://cmpesjsu.slack.com/?' +
+      'redir=%2Fmessages%2FCH6TTGXLG#/">#sce</a></span></p>';
+    let data = undefined;
+
+    try {
+      let x = 0;
+      for (x = 0; x < events.responseData.length; x++) {
```

Comment:
consider using [`array.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) instead

Reply:
refer to update2